code/modules/food/beverages/bottler.dm

```diff
@@ -0,0 +1,413 @@
+			qdel(slots[i])
+			slots[i] = null
+		bottling = 0
+		drink_container.forceMove(loc)
+		updateUsrDialog()
+
+/obj/machinery/bottler/attack_ai(mob/user)
+	attack_hand(user)
+
+/obj/machinery/bottler/attack_ghost(mob/user)
+	attack_hand(user)
+
+/obj/machinery/bottler/attack_hand(mob/user)
+	if(stat & BROKEN)
+		return
+	interact(user)
+
+/obj/machinery/bottler/interact(mob/user)
+	user.set_machine(src)
+	//html ahoy
+	var/dat = ""
+	if(bottling)
+		dat = "<h2>Bottling in process, please wait...</h2>"
+	else
+		dat += "<table border='1' style='width:75%'>"
+		dat += "<tr>"
+		dat += "<th colspan='3'>Containers:</th>"
+		dat += "</tr>"
+		dat += "<tr>"
+		dat += "<td>Glass Bottles: [containers["glass bottle"]]</td>"
+		dat += "<td>Plastic Bottles: [containers["plastic bottle"]]</td>"
+		dat += "<td>Metal Cans: [containers["metal can"]]</td>"
+		dat += "</tr>"
+		dat += "<tr>"
+		if(containers["glass bottle"])
+			dat += "<td><A href='?src=\ref[src];dispense=1'>Dispense</a></td>"
+		else
+			dat += "<td>Out of stock</td>"
+		if(containers["plastic bottle"])
+			dat += "<td><A href='?src=\ref[src];dispense=2'>Dispense</a></td>"
+		else
+			dat += "<td>Out of stock</td>"
+		if(containers["metal can"])
+			dat += "<td><A href='?src=\ref[src];dispense=3'>Dispense</a></td>"
+		else
+			dat += "<td>Out of stock</td>"
+		dat += "</tr>"
+		dat += "</table>"
+		dat += "<hr>"
+		dat += "<table border='1' style='width:75%'>"
+		dat += "<tr>"
+		dat += "<th colspan='4'>Ingredient Tray Contents:</th>"
+		dat += "</tr>"
+
+		dat += "<tr>"
+		for(var/i = 1, i <= slots.len, i++)
+			var/obj/O = slots[i]
+			if(O)
+				dat += "<td>[bicon(O)]<br>[O.name]</td>"
+			else
+				dat += "<td>Tray Empty</td>"
+
+		if(slots[1] && slots[2] && slots[3])
+			dat += "<td><A href='?src=\ref[src];process=1'>Process Ingredients</a></td>"
+		else
+			dat += "<td>Insufficient Ingredients</td>"
+		dat += "</tr>"
+
+		dat += "<tr>"
+		for(var/i = 1, i <= slots.len, i++)
+			if(slots[i])
+				dat += "<td><A href='?src=\ref[src];eject=[i]'>Eject</a></td>"
+			else
+				dat += "<td>N/A</td>"
+		dat += "<td><A href='?src=\ref[src];eject=4'>Eject All</a></td>"
+		dat += "</tr>"
+		dat += "</table>"
+		dat += "<hr>"
+		dat += "<p>Insert three ingredients and press process to create a beverage. You will be able to select a container for the beverage before processing begins.</p>"
+		dat += "<p>Inserting empty bottles and cans, as well as sheets of glass, plastic, or metal will restock the appropriate container supply.</p>"
+	var/datum/browser/popup = new(user, "bottler", "Bottler Menu", 575, 400)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/bottler/Topic(href, href_list)
+	if(..())
+		return 1
+
+	add_fingerprint(usr)
+	usr.set_machine(src)
+
+	if(href_list["process"])
+		var/list/choices = list("Glass Bottle" = 1, "Plastic Bottle" = 2, "Metal Can" = 3, "Cancel" = 4)
+		var/selection = input("Select a container for your beverage.", "Container") in choices
```

Comment:
You don't need a unique "Cancel" option here, you can use `in choices | null` to include a cancel button on the input box that will return null.

Also, you should probably include a check for `bottling` before you start checking the href list, to make sure the machine accepts no input whatsoever while it's bottling.